src/libs/NextStepUtils.ts

```diff
@@ -27,5 +50,270 @@ function parseMessage(messages: Message[] | undefined) {
+                ];
+            }
+
+            break;
+
+        // Generates an optimistic nextStep once a report has been submitted
+        case CONST.REPORT.STATUS_NUM.SUBMITTED: {
+            const verb = isManager ? 'review' : 'approve';
+
+            // Self review & another reviewer
+            optimisticNextStep = {
+                type,
+                title: 'Next Steps:',
+                message: [
+                    {
+                        text: 'Waiting for ',
+                    },
+                    {
+                        text: managerDisplayName,
+                        type: 'strong',
+                    },
+                    {
+                        text: ' to ',
+                    },
+                    {
+                        text: verb,
+                        type: 'strong',
+                    },
+                    {
+                        text: ' %expenses.',
+                    },
+                ],
+            };
+
+            // Another owner
+            if (!isOwner) {
+                optimisticNextStep.message = [
+                    {
+                        text: ownerLogin,
+                        type: 'strong',
+                    },
+                    {
+                        text: ' is waiting for ',
+                    },
+                    {
+                        text: 'you',
+                        type: 'strong',
+                    },
+                    {
+                        text: ' to ',
+                    },
+                    {
+                        text: verb,
+                        type: 'strong',
+                    },
+                    {
+                        text: ' these %expenses.',
+                    },
+                ];
+            }
+
+            break;
+        }
+
+        // Generates an optimistic nextStep once a report has been approved
+        case CONST.REPORT.STATUS_NUM.APPROVED:
+            // Self review
+            optimisticNextStep = {
+                type,
+                title: 'Next Steps:',
+                message: [
+                    {
+                        text: 'Waiting for ',
+                    },
+                    {
+                        text: 'you',
+                        type: 'strong',
+                    },
+                    {
+                        text: ' to ',
+                    },
+                    {
+                        text: 'review',
+                        type: 'strong',
+                    },
+                    {
+                        text: ' %expenses.',
+                    },
+                ],
+            };
+
+            // Another owner
+            if (!isOwner) {
+                optimisticNextStep.title = 'Finished!';
+                optimisticNextStep.message = [
+                    {
+                        text: 'No further action required!',
+                    },
+                ];
+            }
```

Comment:
An approver could also be an admin who still needs to reimburse I think, how would this work in this case? It would show them they are done but they still need to pay it, right?

Reply:
> An approver could also be an admin...
Yes, could be. I think it is always an admin, it is not?

Based on a message that we get from the backend, it should be "No further action....". So, looks like it works properly.

https://github.com/Expensify/App/assets/57314631/4ab28404-16b7-44b2-8274-8cf824102b4f

Reply:
Ok lets go with this and see if some issues will be reported!